jena-arq/Grammar/master.jj

```diff
@@ -327,6 +359,35 @@ void AskQuery() : {}
   SolutionModifier()
 }
 
+void JsonQuery() : {}
+{
+  JsonClause()
+  ( DatasetClause() )*
+  WhereClause()
+  SolutionModifier()
+}
+
+void JsonClause() : { Object o ; String s ; }
+{
+  <JSON> { getQuery().setQueryJsonType() ; }
+  <LBRACE>
+  s = String() < PNAME_NS >
+  (
```

Comment:
Shouldn't PNAME_NS be COLON?  PNAME_NS matches  `foo:` so  "s = String() < PNAME_NS >" matches `"abcd" xyz:`

It may be hard to parse JSON using the SPARQL tokens,in which case using some javacc ability to switch details may be needed.